src/Sulu/Bundle/MediaBundle/Media/Manager/MediaManager.php

```diff
@@ -745,6 +753,10 @@ public function delete($id, $checkSecurity = false)
             );
         }
 
+        if (null !== $this->trashManager) {
+            $this->trashManager->store(MediaInterface::RESOURCE_KEY, $mediaEntity);
+        }
```

Comment:
The original file needs to be backup or moved into private directory / storage.

```php
        $removeOriginalFile = true;
        if (null !== $this->trashManager) {
             $removeOriginalFile = true;
             $this->trashManager->store(MediaInterface::RESOURCE_KEY, $mediaEntity);
        }
```

Reply:
done in https://github.com/sulu/sulu/pull/6218/files#diff-2b8ee69c9133c03ab57c7f6a946d7a4adfcfa726793aac1aa1191079a5b36b78R126 🙂